client/app/components/dynamic-parameters/DateRangeParameter.jsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { includes } from 'lodash';
+import { DYNAMIC_DATE_RANGES } from '@/services/query';
+import { DateRangeInput } from '@/components/DateRangeInput';
+import { DateTimeRangeInput } from '@/components/DateTimeRangeInput';
+import DynamicButton from '@/components/dynamic-parameters/DynamicButton';
+
+import './DynamicParameters.less';
+
+const DYNAMIC_DATE_OPTIONS = [
+  { name: 'Last week',
+    value: 'd_last_week',
+    label: () => DYNAMIC_DATE_RANGES.last_week.value()[0].format('MMM D') + ' - ' +
+                 DYNAMIC_DATE_RANGES.last_week.value()[1].format('MMM D') },
+  { name: 'Last month', value: 'd_last_month', label: () => DYNAMIC_DATE_RANGES.last_month.value()[0].format('MMMM') },
+  { name: 'Last year', value: 'd_last_year', label: () => DYNAMIC_DATE_RANGES.last_year.value()[0].format('YYYY') },
+  { name: 'Last 7 days',
+    value: 'd_last_7_days',
+    label: () => DYNAMIC_DATE_RANGES.last_7_days.value()[0].format('MMM D') + ' - Today' },
+];
```

Comment:
Wdyt of adding "This X" as well?

<img width="340" alt="Screen Shot 2019-07-01 at 8 43 03" src="https://user-images.githubusercontent.com/486954/60413092-4ab50b00-9bdc-11e9-92d4-6d99a0a832d5.png">